apps/token/src/hooks/use-transaction.ts

```diff
@@ -60,7 +60,13 @@ export const useTransaction = (
     });
 
     try {
-      const tx = await performTransaction();
+      const result = performTransaction();
+
+      if (result === null) {
```

Comment:
question (blocking): I don't understand this case? Moroer if this happens won't we end up in an infinite pending state? Or is that desired?

Reply:
This is needed because of the approve function for collateral bridge. See here https://github.com/vegaprotocol/frontend-monorepo/pull/495/files#diff-f9779344e8902f6d965ac153e1f431ffbd102bc4749924cd8ecdd20d6c2f60caR45

We could change the rendering logic if needed to avoid having to the check the result of `performTransaction`

Reply:
You are right about not resetting tx state after the early exit. I've fixed that but perhaps it'll be nicer to do the above

Reply:
Gone ahead and passed in config via prop so we don't need to account for it not being available at tx execution time